### ISSUE TITLE
No reporting unused import on `import * as _ from 'lodash'`

### ISSUE THREAD

**Issue by octref** [Microsoft/TypeScript#30760]
**TypeScript Version:**  3.4.1
**Search Terms:** lodash, unused

**Code**

```ts
import * as _ from 'lodash'
```

```bash
 lodash > tsc --noUnusedLocals index.ts                    
 lodash >
 lodash > echo "import * as lodash from 'lodash'" > index.ts
 lodash > tsc --noUnusedLocals index.ts
index.ts:1:1 - error TS6133: 'lodash' is declared but its value is never read.

1 import * as lodash from 'lodash'
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.

 lodash > 
```

**Expected behavior:** TSC / TSServer reports unused import on lodash

**Actual behavior:** TSC / TSServer doesn't report the error. If you change `_` to something else it does.

**Playground Link:** n/a

**Related Issues:** https://github.com/vuejs/vetur/issues/1063#issuecomment-479705839

**Comment by rafaelfbs** [comment]
This issue is related to #24958 and https://github.com/Microsoft/TypeScript/issues/24552#issuecomment-396641129

The checker ignores identifiers starting with _. So we could add:

```typescript
if (isAmbientModule(declaration) ||
   (isVariableDeclaration(declaration) && isForInOrOfStatement(declaration.parent.parent) || isImportedDeclaration(declaration)) &&
    isIdentifierThatStartsWithUnderScore(declaration.name!) &&
    idText(declaration.name! as Identifier).length > 1) {
    continue;
}
```

**Comment by RyanCavanaugh** [comment]
"Starting with `_` (with any length)" is the intended rule. Enforcing a longer length would be a breaking change for people already writing things like `arr.forEach((_, i) => ...)`.
